### ISSUE TITLE
Table "Getting Started" documentation should mention .info and QTable,Table interaction

### ISSUE THREAD

**Issue by gkanarek** [astropy/astropy#5658]
The "Getting Started" page in the astropy.table docs doesn't mention the `.info` attribute for mixin columns (which is referenced [here](http://docs.astropy.org/en/stable/table/mixin_columns.html#mixin-attributes)). It seems like this should be the standard practice, so it should be noted on the main page.

**Comment by taldcroft** [comment]
@gkanarek - agreed!  Maybe you would want to take a crack at a documentation PR to improve this?

**Comment by mhvk** [comment]
[I had opened a new PR for this -- #5979 -- but @larrybradley correctly pointed out this one existed already, so I'm duplicating my comment here and closing the new PR]

Following discussion in https://github.com/gammapy/gammapy/issues/980 with @cdeil: it would be very helpful to document how to best interact with the various column types in `Table` and `QTable`. In particular, e.g., that by accessing information via `info` (such as `info.name`, `info.description`, `info.meta`), one can ensure that code, if appropriate, can be agnostic to the type of column it is dealing with. 

Similarly, we should perhaps make clearer how one can construct one's own mixin column type using a suitable `info` property.

Milestoning this 2.0 since I think it is important that this be clear.

p.s. It may be best to do this after #5910 and #5841 have been merged, i.e., when `QTable` is essentially equivalent to `Table` except for not supporting masks.

p.p.s. EDIT there may be a somewhat larger issue here, in that for people who really want to use astropy it might be helpful to have some overview description of how things are implemented. This certainly holds for `Quantity`...

**Comment by taldcroft** [comment]
Given time constraints, pushing this off to 3.0...

**Comment by pllim** [comment]
Re-milestone to 3.1?

**Comment by taldcroft** [comment]
I thought policy was to milestone to `future` unless someone is assigned or a PR is open.

**Comment by pllim** [comment]
Perhaps @mhvk could clarify.

**Comment by mhvk** [comment]
I generally agree that no issue should ever have a concrete milestone, unless it is a "release blocker". Apparently, I felt it was at the time, but even that clearly that did not help to get it done, so let's just remove the milestone.

**Comment by bsipocz** [comment]
I remember the same approach as @mhvk to milestoning issues, and actually only have some vague guesses what `future` should be used for.

**Comment by astrofrog** [comment]
We used to milestone all PRs, which is why Future existed. We could get rid of that milestone now?

**Comment by cdeil** [comment]
It's good to have a distinction between "no milestone because not triaged yet" and "no milestone because not required for any specific release, just nice to have" (which probably is what "future" means), no?

**Comment by mhvk** [comment]
I think we should distinguish issues from PRs -- I agree PRs should always be milestoned, and then "future" is useful if, e.g., it is an enhancement that is not fully baked yet.
But I like the "triage" argument for issues too -- empty means it hasn't quite been looked at. The difference with PRs would just be that the outcome for issues should generally be "future" -- I've now put that back here!

So, I guess we need something to document this type of logic...

**Comment by bsipocz** [comment]
Well, I'm still not sold on milestoning the issues as future or a similar equivalent for issues reads as "don't bother" to me, and if we use anything else, we then keep remilestoning them. (Future for PRs has a clear meaning as mentioned above.)

Re triaging: if an issue has a label, it was once definitely looked at by someone. I don't see how adding a meaningless milestone would improve that. Ideally we need to do more serious triaging from time to time, and I don't see how initial milestoning helps achieving that.

**Comment by bsipocz** [comment]
Re "future" milestone for PRs. I think we can get rid of it in agreement of not keeping PRs open as reminders.

 All experimental PRs can get the next major release milestone, and if it's abandoned or going nowhere the bot will close it anyway, otherwise can be moved to the next major release again in 6 months time.

**Comment by mhvk** [comment]
@bsipocz - hmm, yes, quite convincing. Maybe less is more: just get rid of "future"?
